back/Cycle1.WebApi/Cycle1.WebApi/Program.cs

```diff
@@ -1,24 +1,38 @@
-using A3.Lea.Cycle1.WebApi;
+using A3.Lea.Cycle1.WebApi.Extensions;
+using Hellang.Middleware.ProblemDetails.Mvc;
 
+var allowAllOriginsInDev = "_allowAllOriginsInDev";
 var builder = WebApplication.CreateBuilder(args);
+builder.Host.UseLogging();
 
 // Add services to the container.
-builder.Services.AddCycle1Service();
+builder.Services.AddCycle1Services();
+builder.Services.AddProblemDetails();
 builder.Services.AddControllers();
+builder.Services.AddProblemDetailsConventions();
 
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
+builder.Services.AddCors(options => options.AddPolicy(name: allowAllOriginsInDev, builder => builder.WithOrigins("*")));
 
 var app = builder.Build();
 
 // Configure the HTTP request pipeline.
 if (app.Environment.IsDevelopment())
 {
+    app.UseDeveloperExceptionPage();
     app.UseSwagger();
     app.UseSwaggerUI();
+    app.UseCors(allowAllOriginsInDev);
 }
-
+else
+{
+    app.UseProblemDetails();
+    app.UseExceptionHandler("/erreur");
+    app.UseHsts();
+}
+app.UseRouting();
```

Comment:
Why this one ? app.UseRouting ?

Reply:
Aaaaaaaaah je ne sais plus :D bonne question